### ISSUE TITLE
[BUGFIX] Adapt to FlexForm language handling deprecation #1215 

### ISSUE THREAD

**Issue by achim71** [FluidTYPO3/flux#1404]
I'm running into the issue that pi_flexform still returns an empty array. It can be fixed by applying the bugfix used for #1215 for the function readDataArrayFromProvidersOrUsingDefaultMethod in Classes\ViewHelpers\Form\DataViewHelper.php.

The problem occures with flux 8.2.0 / typo3 8.7.1

```
protected static function readDataArrayFromProvidersOrUsingDefaultMethod(array $providers, $record, $field)
    {
        if (0 === count($providers)) {
            $lang = static::getCurrentLanguageName();
            $pointer = static::getCurrentValuePointerName();
            
                $dataArray = static::$configurationService->convertFlexFormContentToArray(
                $record[$field],
                null,
                'lDEF',
                'vDEF'
            );
            
        } else {
            $dataArray = [];
            /** @var ProviderInterface $provider */
            foreach ($providers as $provider) {
                $data = (array) $provider->getFlexFormValues($record);
                $dataArray = RecursiveArrayUtility::merge($dataArray, $data);
            }
        }
        
        return $dataArray;
    }
```

https://github.com/FluidTYPO3/flux/pull/1215

**Comment by achim71** [comment]
Updated to flux 8.2.1 and this is still not fixed.

**Comment by cedricziel** [comment]
Would you mind preparing a Pull Request to speed things up a little?

**Comment by achim71** [comment]
Never did this hope it's correct that way https://github.com/FluidTYPO3/flux/pull/1410

**Comment by cedricziel** [comment]
No worries, we welcome everyone to contribute to the project - thanks in advance! :)

**Comment by cedricziel** [comment]
Closed via #1410
